dockerclient/client.go

```diff
@@ -1079,27 +1079,29 @@ func (e *ClientExecutor) Archive(fromFS bool, src, dst string, allowDownload boo
 		klog.V(5).Infof("Archiving %s %s -> %s from context archive", e.ContextArchive, src, dst)
 		return archiveFromFile(e.ContextArchive, src, dst, excludes, check)
 	}
-	// if the source is an archive, extract it under the destination directory
-	srcFullPath := filepath.Join(e.Directory, src)
-	if allowDownload && isArchivePath(srcFullPath) {
-		f, err := os.Open(srcFullPath)
-		if err != nil {
-			return nil, nil, fmt.Errorf("error opening %q: %v", srcFullPath, err)
-		}
-		klog.V(5).Infof("Extracting %s -> %s from local archive", srcFullPath, dst)
-		transform := func(h *tar.Header, r io.Reader) (data []byte, update bool, skip bool, err error) {
-			h.Name = filepath.Join(dst, h.Name)
-			if h.Typeflag == tar.TypeLink {
-				h.Linkname = filepath.Join(dst, h.Linkname)
+	if false {
+		// if the source is an archive, extract it under the destination directory
+		srcFullPath := filepath.Join(e.Directory, src)
+		if allowDownload && isArchivePath(srcFullPath) {
+			f, err := os.Open(srcFullPath)
+			if err != nil {
+				return nil, nil, fmt.Errorf("error opening %q: %v", srcFullPath, err)
```

Comment:
This will cause a stutter.

Reply:
Tweaking the format string to "open %q: %v".

Reply:
The open failure will include an error with srcFullPath mentioned, if you wrap with srcFullPath, then you will print it twice.

Reply:
That whole section's been removed, so I'm going to call it fixed.